util/languages.py

```diff
@@ -0,0 +1,626 @@
+    ]
+
+    for i in RAW_DATA.split("\n"):
+        (alpha_3, terminologic_code, alpha_2, names,
+         french_names) = i.strip().split("|")
+        names = [x.strip() for x in names.split(";")]
+        if alpha_2:
+            three_to_two[alpha_3] = alpha_2
+            english_names[alpha_2] = names
+            two_to_three[alpha_2] = alpha_3
+        for name in names:
+            english_names_to_three[name.lower()] = alpha_3
+        english_names[alpha_3] = names
+
+    for i in NATIVE_NAMES_RAW_DATA:
+        alpha_2 = i['code']
+        alpha_3 = two_to_three[alpha_2]
+        names = i['nativeName']
+        names = [x.strip() for x in names.split(",")]
+        native_names[alpha_2] = names
+        native_names[alpha_3] = names
+
+    @classmethod
+    def iso_639_2_for_locale(cls, locale):
+        """Turn a locale code into an ISO-639-2 alpha-3 language code."""
+        if '-' in locale:
+            language, place = locale.lower().split("-",1)
+        else:
+            language = locale
+        if cls.two_to_three[language]:
+            return cls.two_to_three[language]
+        elif cls.three_to_two[language]:
+            # It's already ISO-639-2.
+            return language
+        return None
+
+    @classmethod
+    def string_to_alpha_3(cls, s):
+        """Try really hard to convert a string to an ISO-639-2 alpha-3 language code."""
+        if not s:
+            return None
+        s = s.lower()
+        if s in cls.english_names_to_three:
+            # It's the English name of a language.
+            return cls.english_names_to_three[s]
+
+        if "-" in s:
+            s = s.split("-")[0]
+
+        if s in cls.three_to_two:
+            # It's already an alpha-3.
+            return s
+        elif s in cls.two_to_three:
+            # It's an alpha-2.
+            return cls.two_to_three[s]
+
+        return None
+
+    @classmethod
+    def name_for_languageset(cls, languages):
+        if isinstance(languages, basestring):
+            languages = languages.split(",")
+        all_names = []
+        if not languages:
+            return ""
+        for l in languages:
+            normalized = cls.string_to_alpha_3(l)
+            native_names = cls.native_names.get(normalized, [])
+            if native_names:
+                all_names.append(native_names[0])
+            else:
+                names = cls.english_names.get(normalized, [])
+                if not names:
+                    raise ValueError("No native or English name for %s" % l)
+                all_names.append(names[0])
+        if len(all_names) == 1:
+            return all_names[0]
+        return "/".join(all_names)
+
+
+def languages_from_accept(accept_languages):
+    """Turn a list of (locale, quality) 2-tuples into a list of language codes.
+    
+    TODO: parse_accept_language returns named 3-tuples. How is this
+    function actually used?
+    """
```

Comment:
I grepped around and I don't think this function is used at all. There is also no test for it. Can you confirm that I can delet it?

Reply:
I think you're right that it may have become obsolete after that change. `languages_from_accept` is being called in core/util/flask_util in `languages_for_request()` but that function isn't being used anywhere and can be deleted, too.